src/main/java/pl/piekoszek/gorskimatches/equation/EquationRandomizer.java

```diff
@@ -2,14 +2,22 @@
 
 import org.springframework.stereotype.Component;
 
-import java.util.Random;
+import java.util.*;
 
 @Component
 public class EquationRandomizer {
 
+    private final EquationGenerator equationGenerator;
+
+    public EquationRandomizer(EquationGenerator equationGenerator) {
+        this.equationGenerator = equationGenerator;
+    }
+
     public String randomEquation() {
-        String[] randomEquations = {"9+2=6", "6-4=3", "5+7=2", "4+6=4", "5+2=8", "1+2=8", "7-2=4"};
-        int positionInString = new Random().nextInt(randomEquations.length);
-        return randomEquations[positionInString];
+        Map<String, Set<String>> allQuizzesAndSolutions = equationGenerator.getAllQuizzesAndSolutions();
```

Comment:
It's using `Object` and `String.valueOf` should be avoided if possible.
`String.valueOf` woks on any type (Object), so compiler won't tell us, if it's not String

```
var quizzes = new ArrayList<>(equationGenerator.getAllQuizzesAndSolutions().keySet());
return quizzes.get(new Random().nextInt(quizzes.size()));
```